packages/lesswrong/components/seasonal/solsticeSeason/hooks.ts

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useRef } from 'react';
+import * as THREE from 'three';
+import { createDayNightShaderMaterial } from './shader';
+import { CONTRAST_AMOUNT, BRIGHTNESS_BOOST, BRIGHTNESS_ADD, DEFAULT_SUN_POSITION } from './solsticeSeasonConstants';
+import { findMeshWithTexture } from './utils';
+import { PointOfView } from './types';
+
+export const useGlobeDayNightMaterial = (): React.MutableRefObject<any> => {
+  const globeMaterialRef = useRef<any>(null);
+  useEffect(() => {
+    globeMaterialRef.current = createDayNightShaderMaterial(CONTRAST_AMOUNT, BRIGHTNESS_BOOST, BRIGHTNESS_ADD);
+    return () => {
+      if (globeMaterialRef.current && typeof globeMaterialRef.current.dispose === 'function') {
+        globeMaterialRef.current.dispose();
+      }
+    };
+  }, []);
+  return globeMaterialRef;
+};
+
+export const useGlobeReadyEffects = (
+  isGlobeReady: boolean,
+  globeRef: React.MutableRefObject<any>,
+  globeMaterialRef: React.MutableRefObject<any>,
+  dayImageUrl: string | undefined,
+  nightImageUrl: string | undefined,
+  luminosityImageUrl: string | undefined,
+  pov: PointOfView,
+  onReady?: () => void,
+  onTexturesLoaded?: () => void
+) => {
+  const onReadyRef = useRef(onReady);
+  const onTexturesLoadedRef = useRef(onTexturesLoaded);
+  useEffect(() => { onReadyRef.current = onReady; }, [onReady]);
+  useEffect(() => { onTexturesLoadedRef.current = onTexturesLoaded; }, [onTexturesLoaded]);
+
+  useEffect(() => {
+    if (!isGlobeReady || !globeRef.current) return;
+
+    const globeObj = globeRef.current;
+    const globeMesh = findMeshWithTexture(globeObj.scene);
+    if (globeMesh?.material && globeMaterialRef.current) {
+      globeMesh.material = globeMaterialRef.current;
+    }
+
+    const loader = new THREE.TextureLoader();
+    let loadedCount = 0;
+    const expectedTextures = [dayImageUrl, nightImageUrl, luminosityImageUrl].filter(Boolean).length;
+    
+    const checkAllTexturesLoaded = () => {
```

Comment:
I don't love this mechanism but fine, if the library really doesn't give you anything other than callbacks.  (You could convert them to promises, but not obviously worth it here)